src/scripts/main.js

```diff
@@ -1,11 +1,24 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  return new Promise((resolve) => {
+    const handler = (e) => {
+      resolve(
+        `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`,
```

Comment:
The template literal used in the `resolve` function is correctly formatted, but ensure that `element.nodeName` and `element.id` are valid properties of the `element` object. If `element` is not a valid DOM element, this could lead to unexpected results.